### PR TITLE
[DEVOPS-1020] add logs in dev mode and nix improvements

### PR DESCRIPTION
**Pull request by disassembler**
These are some basic changes to help QA with running daedalus with a demo cluster.

Overrides runTimePath to "./" when ran with `yarn dev` so logs are written to `Logs/pub/Daedalus.log` in current directory.
Improves nix shell code to give instructions on how to use.
Minor refactor to support multiple derivations in `shell.nix` using `passthru`.

### CODE REVIEW

File: source/main/utils/getRuntimeFolderPath.js [hunk 1]
```diff
@@ -1,6 +1,12 @@
 import path from 'path';
 
+const isProd = process.env.NODE_ENV === 'production';
+
 export default (platform, env, appName) => {
+  if (!isProd) {
+    return './';
```
Comment (nikolaglumac) [inline source/main/utils/getRuntimeFolderPath.js]: @disassembler shouldn't we use `path.join` here?

**Comment by nikolaglumac** [comment]
@disassembler if I got this correctly once this PR is merged even us (Daedalus developers) will have Daedalus logs saved in the project directory when running Daedalus in dev mode?

File: source/main/utils/getRuntimeFolderPath.js (hunk 1 above)
Comment (disassembler) [reply source/main/utils/getRuntimeFolderPath.js]: what would be combining, since it's just the current directory?

Comment (nikolaglumac) [reply source/main/utils/getRuntimeFolderPath.js]: Ah yes... Sorry for my dumb question... All good :)

**Comment by nikolaglumac** [comment]
@DominikGuzei @cleverca22 please review this PR too! Thanks!

**Pull request merged**